### ISSUE TITLE
[SCHEMA] physio and stim not included for EEG, MEG, iEEG and PET

### ISSUE THREAD

**Issue by Remi-Gau** [bids-standard/bids-specification#890]
`physio` and `stim` suffixes are currently included for [func](https://github.com/bids-standard/bids-specification/blob/master/src/schema/datatypes/func.yaml) and [beh](https://github.com/bids-standard/bids-specification/blob/master/src/schema/datatypes/beh.yaml).

But they are not included for the following data types:

- [x] EEG: https://github.com/bids-standard/bids-specification/blob/master/src/schema/datatypes/eeg.yaml
- [x] iEEG: https://github.com/bids-standard/bids-specification/blob/master/src/schema/datatypes/ieeg.yaml
- [x] MEG: https://github.com/bids-standard/bids-specification/blob/master/src/schema/datatypes/meg.yaml
- [x] PET: https://github.com/bids-standard/bids-specification/blob/master/src/schema/datatypes/pet.yaml
- [x] ASL
- [x] DWI

Implement validation for validator:
- [ ] EEG
- [ ] iEEG
- [ ] MEG
- [ ] PET
- [ ] ASL
- [ ] DWI
- [ ] mri anat

I can't think of a reason for why that's the case, should we add those to the schema ?

**Comment by Remi-Gau** [comment]
pinging @tsalo :wink:

**Comment by Remi-Gau** [comment]
actually the physio page says this:

```
For the template directory name, <datatype> can correspond to any data recording modality, for example func, anat, dwi, meg, eeg, ieeg, or beh.
 ```

So I suspect that PET should not be included but some other datatypes should be included.

This might make make the templates of some section complicated to read: for example the one for anat.

**Comment by Remi-Gau** [comment]
On this topic: @mnoergaard @melanieganz 

should PET support physio / stim recording ?

Or is it like supporting task for PET? See this comment: https://github.com/bids-standard/bids-specification/pull/804#discussion_r638214073

**Comment by effigies** [comment]
It seems like it should. Should also check in about ASL.

**Comment by Remi-Gau** [comment]
@HenkMutsaerts @patsycle

**Comment by HenkMutsaerts** [comment]
That should be fine,as this doesn't significantly change the ASL specification, as far as i understand now? 

We decided however not to support functional ASL yet as this would require all non task ASL (99% of ASL data) to include `task-rest`. @patsycle was there also another reason? 

Physiological simuli are more common in ASL, in the form of an acetazolamide challenge, breath hold, or CO2 challenge. Would this be covered by physio/stim?

**Comment by patsycle** [comment]
@HenkMutsaerts @Remi-Gau Indeed, functional ASL might get implemented later on. If I understand correctly, it was now mainly an issue to get it working in the validator. Marco might remember this.

**Comment by Remi-Gau** [comment]
> Physiological simuli are more common in ASL, in the form of an acetazolamide challenge, breath hold, or CO2 challenge. Would this be covered by physio/stim?

yeah this would be one of the most straightforward way of gas concentration and respiration during your acquisition and it does not need to change anything to the part of ASL spec. We just need mention the physion / stim page that ASL supports it.

**Comment by Remi-Gau** [comment]
> @HenkMutsaerts @Remi-Gau Indeed, functional ASL might get implemented later on. If I understand correctly, it was now mainly an issue to get it working in the validator. Marco might remember this.

If you are referring to the fact that physio files in the `perf` folder should also be also be validated that is correct. This would need to be implemented.

**Comment by Remi-Gau** [comment]
I suggest implementing this in several PRs split up by datatypes with a counterparts on the validator side.

- eeg, meg, ieeg
- dwi
- asl
- pet
- anat (I am struggling to imagine the use case for  physio and stim for anat but I may be missing something)

**Comment by Remi-Gau** [comment]
OK I will wait for the bids-standard/bids-specification#883 to be in before opening PRs on this to avoid potential merge conflicts.

**Comment by Remi-Gau** [comment]
Will actually transfer this to the validator to remember to implement it.
